sys/arm/broadcom/bcm2835/bcm2835_gpio.c

```diff
@@ -289,16 +302,39 @@ bcm_gpio_set_function(struct bcm_gpio_softc *sc, uint32_t pin, uint32_t f)
 static void
 bcm_gpio_set_pud(struct bcm_gpio_softc *sc, uint32_t pin, uint32_t state)
 {
-	uint32_t bank;
-
 	/* Must be called with lock held. */
 	BCM_GPIO_LOCK_ASSERT(sc);
 
-	bank = BCM_GPIO_BANK(pin);
-	BCM_GPIO_WRITE(sc, BCM_GPIO_GPPUD(0), state);
-	BCM_GPIO_WRITE(sc, BCM_GPIO_GPPUDCLK(bank), BCM_GPIO_MASK(pin));
-	BCM_GPIO_WRITE(sc, BCM_GPIO_GPPUD(0), 0);
-	BCM_GPIO_WRITE(sc, BCM_GPIO_GPPUDCLK(bank), 0);
+	if (sc->sc_is2711) { /* BCM2711 */
+		u_int regid = BCM2711_GPIO_REGID(pin);
+		u_int shift = BCM2711_GPIO_SHIFT(pin);
+		uint32_t reg;
+
+		switch (state) {
+		    case BCM2835_PUD_OFF:
+			state = BCM2711_PUD_OFF;
+			break;
+		    case BCM2835_PUD_DOWN:
+			state = BCM2711_PUD_DOWN;
+			break;
+		    case BCM2835_PUD_UP:
+			state = BCM2711_PUD_UP;
+			break;
+		}
```

Comment:
We have an unusual style for `switch` statements in FreeBSD, in which the `case` lines should not be indented. Meaning:
```
switch (state) {
case BCM2835_PUD_OFF:
        state = BCM2711_PUD_OFF;
        break;
case BCM2835_PUD_DOWN:
...
```